packages/mermaid/src/diagrams/xychart/xychartDetector.ts

```diff
@@ -0,0 +1,20 @@
+import type { DiagramDetector, ExternalDiagramDefinition } from '../../diagram-api/types.js';
+
+const id = 'xychart';
+
+const detector: DiagramDetector = (txt) => {
+  return txt.match(/^\s*xychart/i) !== null;
+};
+
+const loader = async () => {
```

Comment:
```suggestion
const loader: DiagramLoader = async () => {
```

Imported from `../../diagram-api/types.js`, so the import would be:

```ts
import type {
  DiagramDetector,
  DiagramLoader,
  ExternalDiagramDefinition,
} from '../../diagram-api/types.js';
```

Reply:
Done